lab-derek/test/year-router-test.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+require('dotenv').config({path: `${__dirname}/./.test.env`});
+
+const expect = require('expect');
+const superagent = require('superagent');
+
+
+const server = require('../lib/server.js');
+const clearDB = require('./lib/clear-db.js');
+const mockYear = require('./lib/mock-year.js');
+
+const API_URL = (process.env.API_URL);
+let tempYear;
+let tempYears = [];
+
+describe('testing api/years', () => {
+  //DONE - TODO: your tests should start your server when they begin and stop your server when they finish
+  before(server.start);
+  after(server.stop);
+  afterEach(clearDB);
+
+  //TODO: write a test to ensure that your api returns a status code of 404 for routes that have not been registered
+
+  describe('testing 404 for unregistered routes', () => {
+    it('should return 404', () => {
+      let data = {name: 2017, dayJan1: 'sun'};
+      return superagent.post(`${API_URL}/api/does-not-exist`)
+      .send(data)
+      .catch(res => {
+        expect(res.status).toEqual(404);
+      });
+    });
+  });
+
+//TODO: POST - test 200, response body like {<data>} for a post request with a valid body
+  describe('testing POST api/years', () => {
+    it('should return a status 200 and year', () => {
+      let data = {name: 2017, dayJan1: 'sun'};
+      return superagent.post(`${API_URL}/api/years`)
+      .send(data)
+      .then(res => {
+        expect(res.status).toEqual(200);
+        expect(res.body.name).toEqual(2017);
+        expect(res.body.dayJan1).toEqual('sun');
+      });
+    });
+    //TODO: POST - test 400, with an invalid request body
+    it('should return a status 400 with invalid request body - name', () => {
+      let data = {name: '2017', dayJan1: 'sun'};
+      return superagent.post(`${API_URL}/api/years`)
+      .send(data)
+      .catch(res => {
+        expect(res.status).toEqual(400);
+      });
+    });
+  //TODO: POST - test 409, with an a conflict for a unique property
+    it('should return a status 409 with invalid request body - name', () => {
+      let data = {name: 2017, dayJan1: 'sun'};
+      superagent.post(`${API_URL}/api/years`)
```

Comment:
You might need to return this first post. Ideally you would use your mock-year schema to just create a resource. You don't really want to make extra requests in a test.